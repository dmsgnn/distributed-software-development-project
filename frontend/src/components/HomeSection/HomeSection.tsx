import { Box, Text, Heading, VStack } from '@chakra-ui/react';
import { StaticImageData } from 'next/image';
import { CSSProperties, FC } from 'react';

type Variant = {
	title: string;
	description: string;
	style: CSSProperties;
};

type Variants = {
	gdpr: Variant;
	flawDetection: Variant;
	collaboration: Variant;
};

const variants: Variants = {
	gdpr: {
		title: 'GDPR Compliance',
		description: 'Use DSEC to be sure that your software is compliant with GDPR laws.',
		style: { backgroundColor: '#00053E', color: 'white' },
		// img: {},
	},
	flawDetection: {
		title: 'Automatic Flaw Detection',
		description: 'Use DSEC to find security flaws in your software that could compromise the safety of your users.',
		style: { backgroundColor: '#BB82F8', color: '#161616' },
		// img: {},
	},
	collaboration: {
		title: 'Collaboration',
		description:
			'DSEC has been designed to enhance collaboration, get everyone in your team working as one â€“ without disrupting their workflow.',
		style: { backgroundColor: '#AD691A', color: 'white' },
		// img: {},
	},
};

interface HomeSectionProps {
	variant: 'gdpr' | 'flawDetection' | 'collaboration';
}

const HomeSection: FC<HomeSectionProps> = ({ variant }) => {
	const { title, description, style } = variants[variant];
	return (
		<Box as="section" sx={commonStyles} style={style}>
			<VStack justifyContent="space-between" h="100%" alignItems="flex-start" maxW="50%">
				<Heading as="h3">{title}</Heading>
				<Text>{description}</Text>
				<div></div>
			</VStack>
		</Box>
	);
};

const commonStyles: CSSProperties = {
	height: 400,
	width: '100%',
	padding: '4rem 6rem',
	borderRadius: 36,
};

export default HomeSection;
