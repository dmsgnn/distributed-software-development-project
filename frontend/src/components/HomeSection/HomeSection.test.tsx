import { axe } from 'jest-axe';

import { act, render } from 'test-utils';
import HomeSection from './HomeSection';

describe('HomeSection', () => {
	it('should matches gdpr snapshot', () => {
		const { asFragment } = render(<HomeSection variant="gdpr" />);

		expect(asFragment()).toMatchSnapshot();
	});

	it('should be accessible (gdpr)', async () => {
		const { container } = render(<HomeSection variant="gdpr" />);

		let results: Awaited<ReturnType<typeof axe>>;
		await act(async () => {
			results = await axe(container);
		});

		expect(results).toHaveNoViolations();
	});

	it('should matches privacy snapshot', () => {
		const { asFragment } = render(<HomeSection variant="collaboration" />);

		expect(asFragment()).toMatchSnapshot();
	});

	it('should be accessible (privacy)', async () => {
		const { container } = render(<HomeSection variant="collaboration" />);

		let results: Awaited<ReturnType<typeof axe>>;
		await act(async () => {
			results = await axe(container);
		});

		expect(results).toHaveNoViolations();
	});

	it('should matches terms snapshot', () => {
		const { asFragment } = render(<HomeSection variant="flawDetection" />);

		expect(asFragment()).toMatchSnapshot();
	});

	it('should be accessible (terms)', async () => {
		const { container } = render(<HomeSection variant="flawDetection" />);

		let results: Awaited<ReturnType<typeof axe>>;
		await act(async () => {
			results = await axe(container);
		});

		expect(results).toHaveNoViolations();
	});
});
