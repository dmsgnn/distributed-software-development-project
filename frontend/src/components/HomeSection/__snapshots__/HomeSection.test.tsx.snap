// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeSection should matches gdpr snapshot 1`] = `
<DocumentFragment>
  <section
    class="chakra-stack css-1r8bz70"
    style="background-color: rgb(0, 5, 62); color: white;"
  >
    <div
      class="chakra-stack css-9fdz3h"
    >
      <h3
        class="chakra-heading albra css-0"
      >
        GDPR Compliance
      </h3>
      <p
        class="chakra-text css-0"
      >
        Use DSEC to be sure that your software is compliant with GDPR laws.
      </p>
      <div />
    </div>
    <img
      alt=""
      data-nimg="1"
      decoding="async"
      height="270"
      loading="lazy"
      role="presentation"
      src="/_next/image?url=%2Fimg.jpg&w=828&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 1x, /_next/image?url=%2Fimg.jpg&w=828&q=75 2x"
      style="color: transparent;"
      width="400"
    />
  </section>
</DocumentFragment>
`;

exports[`HomeSection should matches terms snapshot 1`] = `
<DocumentFragment>
  <section
    class="chakra-stack css-1r8bz70"
    style="background-color: rgb(187, 130, 248); color: rgb(22, 22, 22);"
  >
    <div
      class="chakra-stack css-9fdz3h"
    >
      <h3
        class="chakra-heading albra css-0"
      >
        Automatic Flaw Detection
      </h3>
      <p
        class="chakra-text css-0"
      >
        Use DSEC to find security flaws in your software that could compromise the safety of your users.
      </p>
      <div />
    </div>
    <img
      alt=""
      data-nimg="1"
      decoding="async"
      height="270"
      loading="lazy"
      role="presentation"
      src="/_next/image?url=%2Fimg.jpg&w=828&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 1x, /_next/image?url=%2Fimg.jpg&w=828&q=75 2x"
      style="color: transparent;"
      width="400"
    />
  </section>
</DocumentFragment>
`;
